api-resources.md

```diff
@@ -0,0 +1,414 @@
+
+##### POST
+
+POST to an aggregate job is how the leader steps an aggregate job. The request body is a `struct AggregateJob` where `prepare_steps` contains the leader's current-round prepare messages, and `round` is the number of the current round. The response is a `struct AggregateJob` where `prepare_steps` contains the helper's next-round prepare messages. The response's `round` field will be one more than the request's `round`.
+
+We use a POST here because this request is _not_ idempotent. Suppose the leader sends `POST /tasks/task-id/aggregate_jobs/aggregation-job-id` where the body has `round = 1`, and that this is successfully handled by the helper. Besides generating the response, this will have the side effect of advancing the state in the helper to round 2. If the leader were to resend the `POST /tasks/task-id/aggregate_jobs/aggregation-job-id` with `round = 1`, then the request should be rejected by the helper, because it has advanced to the next round.
+
+This is analogous to DAP-02's `POST /aggregate` with an `AggregateContinueReq`.
+
+###### Context for the `round` field
+
+The `round` field is needed so that the leader can recover from a response being lost. Suppose a 2-round VDAF is being executed, and that the leader does `PUT /tasks/task-id/aggregate_jobs/aggregation-job-id` and that request succeeds, so the helper responds with its first-round prepare messages. Then, the leader does `POST /tasks/task-id/aggregate_jobs/aggregation-job-id` with the first-round broadcast prepare message, but the helper's response gets lost during network transit.
+
+If the helper received the `POST` and advanced its state to the second round, then the leader should do `GET /tasks/task-id/aggregate_jobs/aggregation-job-id` so it can compute the second round broadcast message and then do `POST /tasks/task-id/aggregate_jobs/aggregation-job-id` to have the helper move to the finished state. If the helper did not receive the `POST`, then the leader should re-send the first-round broadcast prepare message.
+
+But in DAP-02, the leader has no way to know what happened in the helper, leaving it unable to recover from this state. Adding a `round` field to the `AggregateJob` message enables the leader to reliably and idempotently find out what state the helper is in by doing `GET /tasks/task-id/aggregate_jobs/aggregation-job-id`, and then taking the appropriate next step.
+
+##### DELETE
+
+Instructs the helper to abandon the aggregate job and allows it to discard all state related to it.
+
+Requiring this would solve https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/241
+
+### Aggregation jobs
+
+Only the helper supports this resource.
+
+#### Path
+
+`/tasks/task-id/aggregation_jobs`
+
+#### Representation
+
+A paginated list of aggregation jobs for the task.
```

Comment:
Given that aggregation job IDs are assigned by the leader, I believe you are right, because the leader should be able to maintain an authoritative list of agg jobs itself instead of relying on the helper.